ekncontent/ekncontent/eknc-domain.c

```diff
@@ -1013,11 +1040,16 @@ ptr_array_to_list_deep (GPtrArray *array)
   GList *list = NULL;
 
   /* If there's no array just return an empty list */
-  if (!array)
+  if (array == NULL)
     return NULL;
 
-  for (guint i = 0; i < array->len; ++i)
-    list = g_list_append (list, g_object_ref (g_ptr_array_index (array, i)));
+  /* Walk the list backwards so we can use prepend() to get constant
+   * time insertion in the results list, instead of having to use
+   * append(), or to use prepend() and then reverse() before returning
+   * it at the end
+   */
+  for (guint i = array->len - 1; i >= 0; i--)
```

Comment:
Here's the crash that you are currently getting; if `array->len == 0` then you are counting backwards from 4294967296...

This worked for me:
```c
  for (guint i = array->len; i > 0; i--)
    list = g_list_prepend (list, g_object_ref (g_ptr_array_index (array, i - 1)));
```
At least this gets it back to the previous state where a window shows up, but no content.

Reply:
Duh. Fixed, and now I'm getting a lovely stack corruption I can debug :-)